code/modules/SCP/SCPs/SCP-2427-3.dm

```diff
@@ -0,0 +1,336 @@
+GLOBAL_LIST_EMPTY(scp2427_3s)
+
+/datum/scp/scp_2427_3
+	name = "SCP-2427-3"
+	designation = "2427-3"
+	classification = EUCLID
+
+/obj/item/natural_weapon/leg_2427_3
+	name = "robotic leg"
+	attack_verb = list("stabbed")
+	hitsound = 'sound/scp/2427/stab.ogg'
+	damtype = BRUTE
+	melee_accuracy_bonus = 200
+	edge = 1
+	force = 36
+
+/mob/living/simple_animal/hostile/scp_2427_3
+	name = "SCP-2427-3"
+	desc = "An amalgamation of exposed wires and robotic parts. It has 4 spider-like legs and a metal mask in place of the 'head'."
+	icon = 'icons/SCP/scp-2427-3.dmi'
+	icon_state = null
+	icon_dead = "dead"
+	pixel_x = -8
+	default_pixel_x = -8
+
+	SCP = /datum/scp/scp_2427_3
+	status_flags = NO_ANTAG
+
+	see_invisible = SEE_INVISIBLE_NOLIGHTING
+	see_in_dark = 7
+
+	maxHealth = 2000
+	health = 2000
+
+	movement_cooldown = 5
+	movement_sound = 'sound/mecha/mechmove04.ogg'
+
+	a_intent = "harm"
+	can_be_buckled = FALSE
+
+	natural_weapon = /obj/item/natural_weapon/leg_2427_3
+	natural_armor = list(
+		melee = ARMOR_MELEE_KNIVES
+		)
+
+	var/satiety = 300
+	/// How much satiety is reduced per tick
+	var/satiety_reduction_per_tick = 0.5
+	/// Upon going to that point or above - the mob goes into is_sleeping stage and is unable to act/speak/move for some time
+	var/max_satiety = 500
+	/// Upon that point, the mob is on rampage until getting above half of max satiety and can attack anything
+	var/min_satiety = 0
+	/// When TRUE - it ignores purity list and can attack anything
+	var/enraged = FALSE
+	var/is_sleeping = FALSE
+	/// If is_sleeping, getting down to that health will wake us up
+	var/wakeup_health = 0
+	var/door_cooldown
+	var/area/spawn_area = null
+	// Upon encountering a mob, they are added in one of these
+	var/list/purity_list = list()
+	var/list/impurity_list = list()
+
+/mob/living/simple_animal/hostile/scp_2427_3/Initialize()
+	GLOB.scp2427_3s |= src
+	spawn_area = get_area(src)
+	add_language(LANGUAGE_ENGLISH, FALSE)
+	return ..()
+
+/mob/living/simple_animal/hostile/scp_2427_3/Destroy()
+	GLOB.scp2427_3s -= src
+	return ..()
+
+/mob/living/simple_animal/hostile/scp_2427_3/Life()
+	. = ..()
+	if(!.)
+		return
+	if(!ckey)
```

Comment:
I don't want to punish AFK/disconnected players.

Reply:
When person disconnects, his client is deleted and mob's client is null.

Reply:
Exactly why it is using ckey and not client.
Since satiety is essentially breach timer, wouldn't be nice to delay it for people that disconnected for a minute.